src/main/java/org/shanerx/tradeshop/utils/Utils.java

```diff
@@ -476,55 +478,66 @@ public ExchangeStatus canExchangeAll(Shop shop, Inventory playerInv, int multipl
         return ExchangeStatus.SUCCESS; //Successfully completed trade
 	}
 
-    //Returns an arraylist of the itemstacks to be removed/added, if it could not get enough of an item, will return index 0 as null and index 1 as item it could not get enough of
-	public ArrayList<ItemStack> getItems(Inventory inventory, List<ShopItemStack> items, int multiplier) {
-        Inventory clone = Bukkit.createInventory(null, inventory.getStorageContents().length);
-        clone.setContents(inventory.getStorageContents());
-        ArrayList<ItemStack> ret = new ArrayList<ItemStack>();
-        int totalCount = 0, currentCount = 0;
-        debugger.log("ShopTradeListener > Inventory Type Being Searched: " + inventory.getType().name(), DebugLevels.TRADE);
-		debugger.log("ShopTradeListener > Inventory Location Being Searched: " + (inventory.getLocation() != null ? inventory.getLocation().toString() : "null"), DebugLevels.TRADE);
-
-		for (ShopItemStack item : items) {
-			totalCount += item.getAmount() * multiplier;
-			int count = item.getAmount() * multiplier;
+	//Returns an arraylist of the itemstacks to be removed/added, if it could not get enough of an item, will return index 0 as null and index 1 as item it could not get enough of
+	public ArrayList<ItemStack> getItems(Inventory inventory, List<ShopItemStack> search, int multiplier) {
+		Map<ItemStack, Integer> storage = new HashMap<>(), found = new HashMap<>();
+		ArrayList<ItemStack> good = new ArrayList<ItemStack>(), bad = new ArrayList<ItemStack>();
 
-			debugger.log("ShopTradeListener > Item Material Being Searched for: " + item.getItemStack().getType().name(), DebugLevels.TRADE);
-			debugger.log("ShopTradeListener > Item count: " + count, DebugLevels.TRADE);
+		debugger.log("ShopTradeListener > Search List: " + search, DebugLevels.TRADE);
```

Comment:
Properly prints the list(this is why the toStrings were added) it isn't printed pretty but is completely readable outside of shulker contents(since they print as a B64)

Reply:
Screenshot of the output that I had put in discord
![unknown](https://user-images.githubusercontent.com/7243280/160692581-a61c0578-bb11-45df-a16f-7a51e6da83ba.png)